pyvista/core/grid.py

```diff
@@ -852,3 +852,37 @@ def extent(self, new_extent: Sequence[int]):
     def to_tetrahedra(self, *args, **kwargs):
         """Cast to a rectangular grid and then convert to tetrahedra."""
         return self.cast_to_rectilinear_grid().to_tetrahedra(*args, **kwargs)
+
+
+class UniformGrid(ImageData):
+    """Deprecated ``UniformGrid`` and renamed to ``ImageData``.
+
+    .. deprecated:: 0.40.0
+        ``pyvista.UniformGrid`` has been deprecated and renamed to ``pyvista.ImageData``.
+
+    Future versions of PyVista will implement ``UnformGrid``
+    as a subclass of ``vtkUniformGrid``, not ``vtkImageData``.
+    """
+
+    def __init__(
+        self,
+        uinput=None,
+        *args,
+        dimensions=None,
+        spacing=(1, 1, 1),
+        origin=(0, 0, 0),
+        deep=False,
+        **kwargs,
+    ):
+        super().__init__(
+            uinput,
+            *args,
+            dimensions=dimensions,
+            spacing=spacing,
+            origin=origin,
+            deep=deep,
+            **kwargs,
+        )
+        warnings.warn(
+            '`UniformGrid` is deprecated. Use `ImageData` instead.', PyVistaDeprecationWarning
+        )
```

Comment:
As also noted in another thread, I'd definitely make this a thin wrapper with `def __init__(self, *args, **kwargs)` to push users to the right type.